File: ruby/rbfind/lib/rbfind/console_color.rb
# frozen_string_literal: true

################################################################################
#
# console_color.rb
#
# Add basic console color methods to String
#
# NOTE: for this to work as an add-on/mixin to String, it must not be in
#       a module
#
################################################################################

module RbFind
  module ConsoleColor
    RESET = 0
    BLACK = 30
    RED = 31
    GREEN = 32
    YELLOW = 33
    BLUE = 34
    MAGENTA = 35
    CYAN = 36
    WHITE = 37
  end
end

class String
  def black
    "\e[0;#{RbFind::ConsoleColor::BLACK}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_black
    "\e[1;#{RbFind::ConsoleColor::BLACK}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def red
    "\e[0;#{RbFind::ConsoleColor::RED}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_red
    "\e[1;#{RbFind::ConsoleColor::RED}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def green
    "\e[0;#{RbFind::ConsoleColor::GREEN}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_green
    "\e[1;#{RbFind::ConsoleColor::GREEN}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def yellow
    "\e[0;#{RbFind::ConsoleColor::YELLOW}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_yellow
    "\e[1;#{RbFind::ConsoleColor::YELLOW}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def blue
    "\e[0;#{RbFind::ConsoleColor::BLUE}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_blue
    "\e[1;#{RbFind::ConsoleColor::BLUE}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def magenta
    "\e[0;#{RbFind::ConsoleColor::MAGENTA}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_magenta
    "\e[1;#{RbFind::ConsoleColor::MAGENTA}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def cyan
    "\e[0;#{RbFind::ConsoleColor::CYAN}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_cyan
    "\e[1;#{ConsoleColor::CYAN}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def white
    "\e[0;#{RbFind::ConsoleColor::WHITE}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def bold_white
    "\e[1;#{RbFind::ConsoleColor::WHITE}m#{self}\e[#{RbFind::ConsoleColor::RESET}m"
  end

  def in_color(console_color)
    case console_color
    in RbFind::ConsoleColor::BLACK
      black
    in RbFind::ConsoleColor::RED
      red
    in RbFind::ConsoleColor::GREEN
      green
    in RbFind::ConsoleColor::YELLOW
      yellow
    in RbFind::ConsoleColor::BLUE
      blue
    in RbFind::ConsoleColor::MAGENTA
      magenta
    in RbFind::ConsoleColor::CYAN
      cyan
    in RbFind::ConsoleColor::WHITE
      white
    end
  end
end

File: dart/dartfind/lib/src/console_color.dart
enum Color { black, red, green, yellow, blue, magenta, cyan, white }

class ConsoleColor {
  static const String reset = '\u001B[0m';
  static const String black = '\u001B[0;30m';
  static const String red = '\u001B[0;31m';
  static const String green = '\u001B[0;32m';
  static const String yellow = '\u001B[0;33m';
  static const String blue = '\u001B[0;34m';
  static const String magenta = '\u001B[0;35m';
  static const String cyan = '\u001B[0;36m';
  static const String white = '\u001B[0;37m';

  static const String boldBlack = '\u001B[1;30m';
  static const String boldRed = '\u001B[1;31m';
  static const String boldGreen = '\u001B[1;32m';
  static const String boldYellow = '\u001B[1;33m';
  static const String boldBlue = '\u001B[1;34m';
  static const String boldMagenta = '\u001B[1;35m';
  static const String boldCyan = '\u001B[1;36m';
  static const String boldWhite = '\u001B[1;37m';

  static String fromColor(Color color) {
    switch (color) {
      case Color.black:
        return black;
      case Color.red:
        return red;
      case Color.green:
        return green;
      case Color.yellow:
        return yellow;
      case Color.blue:
        return blue;
      case Color.magenta:
        return magenta;
      case Color.cyan:
        return cyan;
      case Color.white:
        return white;
    }
  }
}
